src/hooks/useSwapCallback.ts

```diff
@@ -140,7 +112,8 @@ export function useSwapCallback(
   allowedSlippage: number = INITIAL_ALLOWED_SLIPPAGE, // in bips
   recipientAddressOrName: string | null, // the ENS name or address of the recipient of the trade, or null if swap should be returned to sender
 ): { state: SwapCallbackState; callback: null | (() => Promise<string>); error: string | null } {
-  const { account, chainId, library } = useActiveWeb3React()
+  const { account, chainId } = useActiveWeb3React()
+  const { library } = useWeb3React()
```

Comment:
why split library to an other hook?

Reply:
No, we not split to any other hook. Previously, useActiveWeb3React wrap library and return out. I just removed it to make `useActiveWeb3React` only return common things between EVM & Solana (& any future network). If we want something specific from evm web3, use `useWeb3React`. For Solana use `useWallet`.